clippy_lints/src/methods/string_lit_chars_any.rs

```diff
@@ -0,0 +1,80 @@
+use clippy_utils::{
+    diagnostics::span_lint_and_then,
+    is_from_proc_macro, is_trait_method,
+    msrvs::{Msrv, MATCHES_MACRO},
+    path_to_local,
+    source::snippet_opt,
+};
+use itertools::Itertools;
+use rustc_ast::LitKind;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind, HirId, Param, PatKind};
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::STRING_LIT_CHARS_ANY;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'tcx>,
+    recv: &Expr<'_>,
+    param: &'tcx Param<'tcx>,
+    body: &Expr<'_>,
+    msrv: &Msrv,
+) {
+    if !msrv.meets(MATCHES_MACRO) || !is_trait_method(cx, expr, sym::Iterator) {
+        return;
+    };
+
+    // All bindings introduced by `param`
+    let mut bindings = FxHashSet::default();
```

Comment:
Using a `HashSet` to hold a single binding is excessive. The pattern should be a single `PatKind::Binding`. The only other valid pattern is `PatKind::Wild`.